tests/core/test_exp_manager.py

```diff
@@ -0,0 +1,109 @@
+
+import re
+import pytest
+import shutil
+from pathlib import Path
+
+import pytorch_lightning as pl
+from omegaconf.errors import OmegaConfBaseException
+
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager, LoggerMisconfigurationError, CheckpointMisconfigurationError
+
+
+@pytest.fixture
+def cleanup_local_folder():
+    # Asserts in fixture are not recommended, but I'd rather stop users from deleting expensive training runs
+    assert not Path("./lightning_logs").exists()
+    assert not Path("./NeMo_experiments").exists()
+
+    yield
+
+    if Path("./lightning_logs").exists():
+        shutil.rmtree('./lightning_logs')
+    if Path("./NeMo_experiments").exists():
+        shutil.rmtree('./NeMo_experiments')
+
+
+class TestExpManager:
+    @pytest.mark.unit
+    def test_omegaconf(self):
+        """Ensure omegaconf raises an error when an unexcepted argument is passed"""
+        with pytest.raises(OmegaConfBaseException):
+            exp_manager(None, {"unused": 1})
+
+    @pytest.mark.unit
+    def test_trainer_default_logger(self, cleanup_local_folder, tmp_path):
+        """ Test that a trainer with logger errors out with a number of arguments. Test that it works with
+        create_tensorboard_logger set to False
+        """
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"root_dir": str(tmp_path)})
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"explicit_log_dir": str(tmp_path)})
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"resume": True})
+
+        # Check that exp_manager uses trainer.logger
+        log_dir = exp_manager(test_trainer, {"create_tensorboard_logger": False, "create_checkpoint_callback": False})
+        assert log_dir.resolve() == Path("./lightning_logs/0").resolve()
+        assert Path("./lightning_logs").exists()
+        assert Path("./lightning_logs/0").exists()
+
+    @pytest.mark.unit
+    def test_checkpoint_configurations(self, cleanup_local_folder):
+        """ Test that trainer creating modelcheckpoint and asking exp_manager to do it too results in errors, but
+        is error free if only one is asked to do so.
+        """
+        disable_tb_logger = {"create_tensorboard_logger": False}
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+        with pytest.raises(CheckpointMisconfigurationError):  # Fails because both try to create modelcheckpoint
+            exp_manager(test_trainer, disable_tb_logger)
+
+        # Should success without error
+        exp_manager(test_trainer, {"create_checkpoint_callback": False}.update(disable_tb_logger))
```

Comment:
python dict.update() returns None. Does this actually perform the correct test with default settings of exp_manager()?

Reply:
Oh good point. Thanks, I'll fix that